pkg/ingester/user_metrics_metadata.go

```diff
@@ -47,15 +47,15 @@ func (mm *userMetricsMetadata) add(metric string, metadata *mimirpb.MetricMetada
 		// Verify that the user can create more metric metadata given we don't have a set for that metric name.
 		if err := mm.limiter.AssertMaxMetricsWithMetadataPerUser(mm.userID, len(mm.metricToMetadata)); err != nil {
 			mm.metrics.discardedMetadataPerUserMetadataLimit.WithLabelValues(mm.userID).Inc()
-			return makeLimitError(mm.limiter.FormatError(mm.userID, err))
+			return formatMaxMetadataPerUserError(mm.limiter, mm.userID)
 		}
 		set = metricMetadataSet{}
 		mm.metricToMetadata[metric] = set
 	}
 
 	if err := mm.limiter.AssertMaxMetadataPerMetric(mm.userID, len(set)); err != nil {
 		mm.metrics.discardedMetadataPerMetricMetadataLimit.WithLabelValues(mm.userID).Inc()
-		return makeMetricLimitError(labels.FromStrings(labels.MetricName, metric), mm.limiter.FormatError(mm.userID, err))
+		return formatMaxMetadataPerMetricError(mm.limiter, labels.FromStrings(labels.MetricName, metric), mm.userID)
```

Comment:
A concern I have with this is the implicit assumption that  `mm.limiter.AssertMaxMetadataPerMetric` returns `errMaxMetadataPerMetricLimitExceeded`. By contract (doc string), it can return "an error", there's no guarantee of the kind of error. I.e., the assumption may break in the future.